src/Symfony/Bundle/WebProfilerBundle/EventListener/TurboDriveCspListener.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\WebProfilerBundle\EventListener;
+
+use Symfony\Component\DependencyInjection\Attribute\When;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+/**
+ * Show the Web Debug Toolbar (WDT) when navigating via Turbo Drive and when a strict Content-Security-Policy is set.
+ * This is done by reusing WDT's nonces.
+ * This service is registered only when TurboBundle is installed.
+ *
+ * @author David Petrásek <davidpetrasek@hotmail.cz>
+ */
+class TurboDriveCspListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly bool $isDebug,
+    ) {
+    }
+
+    public function onKernelResponse(ResponseEvent $event): void
+    {
+        if (!$this->isDebug) {
+            return;
+        }
+        if (!$event->isMainRequest()) {
+            return;
+        }
+
+        $request = $event->getRequest();
+        $routeName = $request->get('_route');
+
+        if ('_wdt' === $routeName) {
+            return;
+        }
+
+        if ($request->headers->has('X-Turbo-Request-Id')) {
+            return;
+        } elseif ($request->headers->has('Turbo-Frame')) {
+            return;
+        } elseif ('turbo_stream' === $request->getPreferredFormat()) {
+            return;
+        }
+
+        $response = $event->getResponse();
+
+        $csp = $response->headers->get('Content-Security-Policy');
+        if (!$csp) {
+            return;
+        }
+
```

Comment:
Maybe here (especially when coded in this bundle) you could check if the content of the response contains the toolbar (or it wont be necessary, right?) 

You could register the Listener _after_ the [WebDebugToolbarListener](https://github.com/symfony/symfony/blob/7.3/src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php) (priority -128) and inject the script only if you detect the [toolbar script](https://github.com/symfony/symfony/blob/7.3/src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar_js.html.twig) in the rendered HTML. This could maybe be done with a text search `<div id="sfwdt`, or `sf-toolbar` (after the other checks)..